File: backend/src/main/java/com/distributrack/service/AnalyticsService.java
package com.distributrack.service;

import com.distributrack.dto.response.AnalyticsResponse;
import com.distributrack.dto.response.DeliveryAnalyticsResponse;
import com.distributrack.dto.response.InventoryAnalyticsResponse;
import com.distributrack.dto.response.PaymentAnalyticsResponse;
import com.distributrack.dto.response.SalesAnalyticsResponse;

import java.time.LocalDate;

/**
 * Business analytics. Access is restricted to business roles
 * (SUPER_ADMIN / OWNER / MANAGER) by SecurityConfig; nothing here is
 * exposed to SHOPKEEPER / DELIVERY_BOY / SALESMAN.
 *
 * All figures come from real repository data — no mock statistics.
 * Revenue follows the established rule: only DELIVERED/COMPLETED orders
 * count, each order exactly once.
 */
public interface AnalyticsService {

    /** KPI overview across orders, payments, inventory and deliveries. */
    AnalyticsResponse getAnalytics();

    /** Sales trend, top products, top shops + status distributions. */
    SalesAnalyticsResponse getSalesAnalytics(LocalDate from, LocalDate to);

    PaymentAnalyticsResponse getPaymentAnalytics();

    DeliveryAnalyticsResponse getDeliveryAnalytics();

    InventoryAnalyticsResponse getInventoryAnalytics();
}

File: frontend/src/services/api/analyticsService.ts
import { axiosInstance } from "@/services/api/axiosInstance";
import type {
  AnalyticsResponse,
  DeliveryAnalyticsResponse,
  InventoryAnalyticsResponse,
  PaymentAnalyticsResponse,
  SalesAnalyticsResponse,
} from "@/types/analytics.types";

/**
 * Analytics API client. All endpoints are SA/OWNER/MANAGER only
 * (SecurityConfig); values come from the real backend.
 *
 * Backend reference: AnalyticsController at /api/analytics.
 * `from`/`to` are ISO dates (YYYY-MM-DD).
 */
export const analyticsService = {
  async getOverview(): Promise<AnalyticsResponse> {
    const response = await axiosInstance.get<AnalyticsResponse>(
      "/analytics/overview",
    );
    return response.data;
  },

  async getSales(params?: { from?: string; to?: string }): Promise<SalesAnalyticsResponse> {
    const response = await axiosInstance.get<SalesAnalyticsResponse>(
      "/analytics/sales",
      { params },
    );
    return response.data;
  },

  async getPayments(): Promise<PaymentAnalyticsResponse> {
    const response = await axiosInstance.get<PaymentAnalyticsResponse>(
      "/analytics/payments",
    );
    return response.data;
  },

  async getDeliveries(): Promise<DeliveryAnalyticsResponse> {
    const response = await axiosInstance.get<DeliveryAnalyticsResponse>(
      "/analytics/deliveries",
    );
    return response.data;
  },

  async getInventory(): Promise<InventoryAnalyticsResponse> {
    const response = await axiosInstance.get<InventoryAnalyticsResponse>(
      "/analytics/inventory",
    );
    return response.data;
  },
};
